gobblin-utility/src/main/java/org/apache/hadoop/fs/FileSystemAdapter.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.fs;
+
+import java.io.IOException;
+
+
+public class FileSystemAdapter {
+  public static void rename(FileSystem fs, Path src, Path dst, Options.Rename options) throws IOException {
```

Comment:
`FileContext` is introduced to handle the protected problem of rename : http://hadoop.apache.org/docs/current/api/org/apache/hadoop/fs/FileSystem.html#rename-org.apache.hadoop.fs.Path-org.apache.hadoop.fs.Path-org.apache.hadoop.fs.Options.Rename...-

Reply:
Yeah, but in FsDataWriter, when we try to use fc to do rename and fs to check the existence, underlying they are using different dfsClients, and we may face inconsistency issue. So we try to use the same dsfClient to do rename here.